packages/nodes-base/nodes/EmailNotification/services.ts

```diff
@@ -0,0 +1,47 @@
+import {Hasura, N8nRequest} from '../Gllue/GenericFunctions';
+import moment = require('moment');
+import {SENDGRID_ACTIONS} from './constants';
+
+class EmailAPI extends Hasura {
+	resource = 'email';
+}
+
+export class GetEmailByTrackIdAPIEndpoint extends EmailAPI {
+	action = 'by-track-id';
+
+	getUrl(id?: string) {
+		return `${this.apiHost}/${this.resource}/${this.action}/${id}`;
+	}
+}
+
+class UpdateEmailByIdAPIEndpoint extends EmailAPI {
+	action = 'update';
+}
+
+export class EmailNotificationService {
+	getEmailByTrackIdEndpoint: GetEmailByTrackIdAPIEndpoint;
+	updateEmailById: UpdateEmailByIdAPIEndpoint;
+
+	constructor(request: N8nRequest) {
+		this.getEmailByTrackIdEndpoint = new GetEmailByTrackIdAPIEndpoint(request);
+		this.updateEmailById = new UpdateEmailByIdAPIEndpoint(request);
+	}
+
+	async getEmailByTrackId(id: string) {
+		return await this.getEmailByTrackIdEndpoint.get(id);
+	}
+
+	buildDataByAction(action: string, datetime: number) {
+		const timestamp = moment.unix(datetime).format();
+		return action === 'click' ? {is_clicked: true, clicked_at: timestamp} :
+			{is_opened: true, opened_at: timestamp};
+	}
```

Comment:
I would suggest to swap the names of of the variables because we are converting timestamp value to datetime;
the input parameter should be `timestamp`, and the converted timestamp should be `datetime`

Reply:
https://github.com/Top20Talent/n8n/pull/9/commits/2acf237ce9f515190ae47f5223722d040633bf27